webrender/src/renderer.rs

```diff
@@ -497,120 +500,115 @@ pub(crate) mod desc {
         ],
     };
 
-    pub const GPU_CACHE_UPDATE: VertexDescriptor = VertexDescriptor {
-        vertex_attributes: &[
-            VertexAttribute {
-                name: "aPosition",
-                count: 2,
-                kind: VertexAttributeKind::U16Norm,
-            },
-            VertexAttribute {
-                name: "aValue",
-                count: 4,
-                kind: VertexAttributeKind::F32,
-            },
-        ],
-        instance_attributes: &[],
-    };
-
-    pub const VECTOR_STENCIL: VertexDescriptor = VertexDescriptor {
-        vertex_attributes: &[
-            VertexAttribute {
-                name: "aPosition",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-        ],
-        instance_attributes: &[
-            VertexAttribute {
-                name: "aFromPosition",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aCtrlPosition",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aToPosition",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aFromNormal",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aCtrlNormal",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aToNormal",
-                count: 2,
-                kind: VertexAttributeKind::F32,
-            },
-            VertexAttribute {
-                name: "aPathID",
-                count: 1,
-                kind: VertexAttributeKind::U16,
-            },
-            VertexAttribute {
-                name: "aPad",
-                count: 1,
-                kind: VertexAttributeKind::U16,
-            },
-        ],
-    };
+    cfg_if! {
+        if #[cfg(feature = "gleam")] {
+            pub const GPU_CACHE_UPDATE: VertexDescriptor = VertexDescriptor {
```

Comment:
The reason why I moved these VertexDesciptors in a `cfg_if` block is, that they are used by features which are not currently supported by our `gfx` backend and wanted to remove the unused variable warnings. `GPU_CACHE_UPDATE` is related to the `Scatter` variant of the `CacheBus` which we don't support yet.  `VECTOR_STENCIL` and `VECTOR_COVER` are for the `pathfinder` feature which AFAIK depends on OpenGL.

Reply:
in order to minimize the changes, let's just mark those with `[cfg_attr(feature = "gfx-hal", allow(unused_something))]` instead of wrapping into `cfg_if`